components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -356,6 +373,18 @@ class NewTabPage extends React.Component<Props, State> {
     this.props.actions.setGeminiHideBalance(hide)
   }
 
+  onTotalPriceOptIn = () => {
+    this.props.actions.onTotalPriceOptIn()
+  }
+  onBtcPriceOptIn = async () => {
+    await Promise.all([
+      this.fetchCryptoDotComTickerPrices(['BTC']),
+      this.fetchCryptoDotComLosersGainers()
+    ])
+    this.props.actions.onBtcPriceOptIn()
+    this.props.actions.onCryptoDotComInteraction()
```

Comment:
Same comment re: a lot of actions going on here all at the same time and an interesting pattern of waiting for something to finish before calling some actions. Can this likewise just be all handled in a single action?